### PR TITLE
boards: mchp: Add option to switch GPIO bank voltage

### PR DESCRIPTION
**Pull request by albertofloyd**
Add KConfig board support to match board jumper settings
Failing to update bank selection will affect GPIOs tied to VTR3

Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

### CODE REVIEW

File: soc/arm/microchip_mec/mec1501/Kconfig.soc [hunk 1]
```diff
@@ -66,3 +66,9 @@ config SOC_MEC1501_EXT_32K_PARALLEL_CRYSTAL
 
 	  Say n if the crystal is connected single ended to
 	  the XTAL2 pin or a 32KHz square wave is on XTAL2.
+
+config SOC_MEC1501_VTR3_1_8V
+	bool "VTR3 power rail is tied to 1.8V"
+	default n
```
Comment (ulfalizer) [inline soc/arm/microchip_mec/mec1501/Kconfig.soc]: `bool` symbols implicitly default to `n`, so can remove this.

Generates nicer docs without the `default n` too ("Implicitly defaults to n" instead of "n is <bunch of dependencies>").

File: soc/arm/microchip_mec/mec1501/Kconfig.soc [hunk 2]
```diff
@@ -66,3 +66,9 @@ config SOC_MEC1501_EXT_32K_PARALLEL_CRYSTAL
...
+	help
+	  Set this is if VTR3 power sourcejumper in the board is changed.
```
Comment (ulfalizer) [inline soc/arm/microchip_mec/mec1501/Kconfig.soc]: Missing newline at end of line.

**Review by ulfalizer — CHANGES_REQUESTED** [review]

File: boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig [hunk 3]
```diff
@@ -7,8 +7,9 @@
 CONFIG_ARM=y
 CONFIG_SOC_MEC1501_HSZ=y
 CONFIG_SOC_SERIES_MEC1501X=y
+CONFIG_SOC_MEC1501_VTR3_1_8V=y
 CONFIG_BOARD_MEC1501MODULAR_ASSY6885=y
-CONFIG_RTOS_TIMER=n
+CONFIG_RTOS_TIMER=y
```
Comment (galak) [inline boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig]: This change to RTOS_TIMER isn't related.  Should be at least a separate commit if not PR.

**Review by galak — DISMISSED** [review]
Need to deal w/RTOS_TIMER change in a separate commit.

**Comment by albertofloyd** [comment]
@galak , @ioannisg  please re-visit

**Review by ulfalizer — APPROVED** [review]
Kconfig nits fixed.

File: boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig (hunk 3 above)
Comment (albertofloyd) [reply boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig]: Removed this from this PR, will enable RTOS driver into another PR that enables other drivers as well.

**Review by franciscomunoz — APPROVED** [review]

**Review by scottwcpg — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]

**Comment by andrewboie** [comment]
@galak can you look at this one again?

**Review by dcpleung — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]
